src/App.tsx

```diff
@@ -1,61 +1,129 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [enteredTitle, setEnteredTitle] = useState(true);
+  const [choosedUser, setChoosedUser] = useState(true);
+
+  const getTodoId = Math.max(...visibleTodos.map(todo => todo.id)) + 1;
+
+  const addTodo = (id: number) => {
+    if (getUserById(id) && title.trim()) {
+      const foundedUser = {
+        id: getTodoId,
+        userId,
+        title,
+        completed: false,
+        user: getUserById(userId),
+      };
+
+      setVisibleTodos(prevTodos => {
+        return [
+          ...prevTodos,
+          foundedUser,
+        ];
+      });
+
+      setUserId(0);
+      setTitle('');
+      setEnteredTitle(true);
+      setChoosedUser(true);
```

Comment:
I'm not sure, but maybe you don't need these setters here, if you set "true" inside the form in "onChange".

Reply:
Here I reset the values to default) But U`r right, everything is OK without these actions)

Reply:
I did exactly the same myself. And then I found that without it, it also works))